src/index.ts

```diff
@@ -1,44 +1,99 @@
+ *
+ * // Execute a RUM session
+ * init(settingsBuilder.toSettings())
+ *     .then(result => {
+ *         // `result` contains the results from the RUM session after
+ *         // completion
+ *     })
+ * ```
+ *
+ * @packageDocumentation
+ */
+import { ClientSettings, ExecutableContainer, SessionResult } from "./@types"
+import whenReady from "./util/loadWhenDocumentReady"
 
-export function init(settings: PageSettings): void {
-    loadWhenReady(() => {
-        console.log('Document loaded and ready')
-        settings = settings || DefaultPageSettings
+/**
+ * Called by tag owner code to initialize a RUM session, either immediately or
+ * after some delay.
+ *
+ * @remarks
+ * Waits for the page to load before processing.
+ *
+ * @param settings Specifies settings affecting client behavior. These are
+ * determined by the tag owner at runtime, so may be used to specify page-level
+ * overrides to general defaults.
+ */
+export default function init(settings: ClientSettings): Promise<SessionResult> {
+    return whenReady().then(() => {
         if (settings.preConfigStartDelay) {
-            setTimeout(startLater(settings), settings.preConfigStartDelay)
-        } else {
-            start(settings)
+            return startLater(settings.preConfigStartDelay, settings)
         }
+        return start(settings)
     })
 }
 
-function startLater(settings: PageSettings): () => void {
-    return function () {
-        start(settings)
-    }
+/**
+ * Called internally if a non-zero {@link ClientSettings.preConfigStartDelay}
+ * setting has been specified. Calls {@link start} after the delay.
+ *
+ * @param delay The approximate time to wait (in milliseconds).
+ * @param settings The settings object passed to {@link init}.
+ */
+function startLater(
+    delay: number,
+    settings: ClientSettings,
+): Promise<SessionResult> {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            start(settings).then((result) => resolve(result))
+        }, delay)
+    })
 }
 
-function start(settings: PageSettings): void {
-    const promises = settings.providers
-        .filter(p => p.testUserAgent())
-        .map(p => p.fetchConfig())
-    if (promises.length) {
-        Promise.all(promises)
-            .then(providerConfigs => {
-                const temp = <[Provider, unknown, unknown[]][]>providerConfigs
-                for (let config of temp) {
-                    let provider: Provider = config[0]
-                    for (let task of config[2]) {
-                        provider.executeTask(config[1], task)
-                    }
-                }
-            })
-            .catch(error => {
-                console.error(error)
-            })
-    } else {
-        console.log('No providers promised to make config calls')
-    }
+/**
+ * Called immediately by {@link init} if no
+ * {@link ClientSettings.preConfigStartDelay} setting has been specified.
```

Comment:
Note: There is a lot happening in this function, we should consider fleshing out this description to explain what it does a bit more. Something like:
```suggestion
 * Starts the RUM session by consuming all provider configurations and converting them to 
 * tasks using the following logic: and expanding them into tasks 
 * 
 * 1) Asynchronously fetches each provider session config and wait for them all to settle.
 * 2) Filter provider config results to only those which fulfilled successfully.
 * 3) For each provider set the configuration and expand the tasks.
 * 4) Process the expanded tasks.
 *  
 *Called immediately by {@link init} if no
 * {@link ClientSettings.preConfigStartDelay} setting has been specified.
```